controllers/configmap.go

```diff
@@ -0,0 +1,117 @@
+package controllers
+
+import (
+	"context"
+	skiperatorv1alpha1 "github.com/kartverket/skiperator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"encoding/json"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+)
+
+//+kubebuilder:rbac:groups=skiperator.kartverket.no,resources=applications,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+
+
+type Config struct {
+	Type string `json:"type"`
+	Audience string `json:"audience"`
+	ServiceAccountImpersonationUrl string `json:"service_account_impersonation_url"`
+	SubjectTokenType string `json:"subject_token_type"`
+	TokenUrl string `json:"token_url"`
+	CredentialSource CredentialSource `json:"credential_source"`
+}
+type CredentialSource struct {
+	File string `json:"file"`
+}
+
+type ConfigMapReconciler struct {
+	client client.Client
+	scheme *runtime.Scheme
+	recorder record.EventRecorder
+}
+
+func (r *ConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.client = mgr.GetClient()
+	r.scheme = mgr.GetScheme()
+	r.recorder = mgr.GetEventRecorderFor("configmap-controller")
+
+	return newControllerManagedBy[*skiperatorv1alpha1.Application](mgr).
+		For(&skiperatorv1alpha1.Application{}).
+		Owns(&corev1.ConfigMap{}).
+		Complete(r)
+}
+
+func (r *ConfigMapReconciler) Reconcile(ctx context.Context, application *skiperatorv1alpha1.Application) (reconcile.Result, error) {
+	application.FillDefaults()
+
+	gcpAuthConfigMapName := application.Name + "-gcp-auth"
+	gcpIdentityConfigMap := corev1.ConfigMap{}
+	gcpAuthConfigMap := corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: application.Namespace, Name: gcpAuthConfigMapName}}
+	
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: "skiperator-system", Name: "gcpidentityconfig"}, &gcpIdentityConfigMap)
+	if errors.IsNotFound(err) {
+		r.recorder.Eventf(
+			application,
+			corev1.EventTypeWarning, "Missing",
+			"Cannot find configmap named gcpidentityconfig in namespace skiperator-system",
+		)
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+	
+	_, err = ctrlutil.CreateOrPatch(ctx, r.client, &gcpAuthConfigMap, func() error {
```

Comment:
Try putting the variables next to where they are used to separate concerns by usage
```suggestion
	gcpIdentityConfigMap := corev1.ConfigMap{}
	err := r.client.Get(ctx, types.NamespacedName{Namespace: "skiperator-system", Name: "gcpidentityconfig"}, &gcpIdentityConfigMap)
	if errors.IsNotFound(err) {
		r.recorder.Eventf(
			application,
			corev1.EventTypeWarning, "Missing",
			"Cannot find configmap named gcpidentityconfig in namespace skiperator-system",
		)
	} else if err != nil {
		return reconcile.Result{}, err
	}
	
	gcpAuthConfigMapName := application.Name + "-gcp-auth"
	gcpAuthConfigMap := corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: application.Namespace, Name: gcpAuthConfigMapName}}
	_, err = ctrlutil.CreateOrPatch(ctx, r.client, &gcpAuthConfigMap, func() error {
```

Reply:
Moved closer to action.